src/generate_transactions.rs

```diff
@@ -118,24 +55,123 @@ pub async fn send_continuous_transactions(
 pub async fn chain_of_transfers(
     transaction_manager: TransactionManager,
     num_transactions: usize,
-    amount: U256,
+    mut transfer_amount: U256,
 ) -> Result<()> {
     let gas = 1.2e13 as u64;
     let mut next_tx_manager = transaction_manager.clone();
 
-    for i in 0..num_transactions {
-        let transfer_amount: U256 = amount - ((i + 1) as u64 * gas);
+    for i in 1..=num_transactions {
         if transfer_amount <= (1e14 as u128).into() {
```

Comment:
This amount should probably be `gas`.
```suggestion
        if transfer_amount <= gas.into() {
```